src/Arcus.Observability.Correlation/DefaultCorrelationInfoAccessor.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System.Threading;
+
+namespace Arcus.Observability.Correlation
+{
+    /// <summary>
+    /// Default <see cref="ICorrelationInfoAccessor"/> implementation to access the <see cref="Correlation.CorrelationInfo"/> in the current context.
+    /// </summary>
+    public class DefaultCorrelationInfoAccessor : ICorrelationInfoAccessor
+    {
+        private static readonly AsyncLocal<CorrelationInfo> CorrelationInfoLocalData = new AsyncLocal<CorrelationInfo>();
```

Comment:
Is this the idea to get rid of the current web api approach and use this instead or?

Reply:
This is the default accessor for I think the messaging repo.
The web app can implement it's own accessor and use the HTTP request features instead.

Reply:
Just as it uses now, only now instead of the `HttpCorrelationInfo` we have there, we'll have a `HttpCorrelationInfoAccessor`.

Reply:
Sounds good to me! Only concern I have is that correlation info for messaging is defined on a pre-message basis so does it work with this appraoch?

Reply:
I was thinking that we could maybe set the correlation info at the same place where we call `message.GetMessageCorrelationInfo()` in the message pump.
We could retrieve the `ICorrelationInfo<MessageCorrelationInfo>` from the `.ServiceProvider` property in the message pump and set the correlation.

Reply:
That's ok for me, but I would do it after calling GetMessageCorrelationInfo which is probably what you are suggesting as well?

Reply:
Yes, indeed. After. Ok, we're on the same page.